virtual_rainforest/models/abiotic/atmospheric_co2.py

```diff
@@ -0,0 +1,235 @@
+r"""The ``models.abiotic.atmospheric_co2`` module calculates the within- and below-
+canopy CO\ :sub:`2`\ profile for the Virtual Rainforest.
+
+Based on external inputs, the module interpolates an initial vertical
+CO\ :sub:`2`\ profile which
+is then modified by plant net carbon assimilation and soil and animal respiration and
+vertically mixed based on the wind profiles above, within, and below the canopy (the
+mixing is currently not implemented).
+
+TODO cross-check input variable names with other modules
+TODO update indexing vertical dimension
+"""  # noqa: D205, D415
+
+from dataclasses import dataclass
+
+import numpy as np
+import xarray as xr
+from xarray import DataArray
+
+from virtual_rainforest.core.data import Data
+
+
+@dataclass
+class CO2Constants:
+    r"""CO\ :sub:`2`\ constants class."""
+
+    ambient_co2_default: float = 400
+    r"""Default ambient CO\ :sub:`2`\ concentration, [ppm]"""
+
+
+def calculate_co2_profile(
+    data: Data,
+    atmosphere_layers: int,  # from config?
+    initialisation_method: str = "homogenous",
+    mixing: bool = False,
+    co2_const: CO2Constants = CO2Constants(),
+) -> DataArray:
+    r"""Calculate CO\ :sub:`2`\ profile.
+
+    This function takes a :class:`~virtual_rainforest.core.data.Data` object as ``data``
+    argument that contains the following variables:
+
+    * atmospheric CO\ :sub:`2`\
+    * plant net CO\ :sub:`2`\ assimilation
+    * soil respiration
+    * animal respiration
+
+    Args:
+        data: A Virtual Rainforest Data object.
+        atmosphere_layers: number of atmosphere layers for which CO\ :sub:`2`\
+            concentration is calculated
+        initialisation_method: interpolation method, default copies ambient
+            CO\ :sub:`2`\ to all vertical levels
+        mixing: flag if mixing is true or false
+
+    Returns:
+        vertical profile of CO\ :sub:`2`\ concentrations, [ppm]
+    """
+
+    # check if atmospheric CO2 concentration in data
+    if "atmospheric_co2" not in data:
+        ambient_atmospheric_co2 = DataArray(
+            np.repeat(
+                co2_const.ambient_co2_default,
+                len(data.grid.cell_id),
+            ),
```

Comment:
This has to be a 1D array as it creates an input co2 concentration for each cell if not provided in data, so it is correct. This might become outdated if I drop the check discussed in the comment above.

Reply:
Right! Sorry - this is the ambient atmospheric (above canopy?) CO2, so is not expected to vary except possibly spatially, which is what you have.

Reply:
I changed the name from `ambient_co2` to `atmospheric_co2_topofcanopy` to make that clearer